src/main/java/graphql/execution/ValuesResolverConversion.java

```diff
@@ -95,47 +106,89 @@ static Object externalValueToInternalValue(GraphqlFieldVisibility fieldVisibilit
                                                GraphQLInputType type,
                                                GraphQLContext graphqlContext,
                                                Locale locale) {
-        return externalValueToInternalValueImpl(fieldVisibility, type, externalValue, graphqlContext, locale);
+        InputInterceptor inputInterceptor = graphqlContext.get(InputInterceptor.class);
+        return externalValueToInternalValueImpl(
+                inputInterceptor,
+                fieldVisibility,
+                type,
+                externalValue,
+                graphqlContext,
+                locale);
     }
 
     @Nullable
-    static Object valueToInternalValueImpl(InputValueWithState inputValueWithState, GraphQLType type, GraphQLContext graphqlContext, Locale locale) {
+    static Object valueToInternalValueImpl(
+            InputInterceptor inputInterceptor,
+            InputValueWithState inputValueWithState,
+            GraphQLInputType inputType,
+            GraphQLContext graphqlContext,
+            Locale locale
+    ) {
```

Comment:
GraphQLType type was always wrong.  Ths is ALWAYS input types - always

We cast at certain places to treat it as a input type and some times we just do instanceof - BUT it should have always been a `GraphQLInputType` and now it is